lib/nostr.js

```diff
@@ -16,24 +20,52 @@ export const DEFAULT_CROSSPOSTING_RELAYS = [
   'wss://nostr.mutinywallet.com/',
   'wss://relay.mutinywallet.com/'
 ]
+export const DEFAULT_NIP46_RELAYS = [
+  'wss://relay.damus.io/',
+  'wss://relay.snort.social/'
+]
 
 export class Relay {
-  constructor (relayUrl) {
-    const ws = new WebSocket(relayUrl)
+  constructor (relayUrl, reconnectOnClose = false) {
+    this.url = relayUrl
+    this.subs = []
+    this.reconnectOnClose = reconnectOnClose
+    this._closed = false
+    this.error = undefined
+    this.reconnect()
+  }
 
-    ws.onmessage = function (msg) {
+  reconnect () {
```

Comment:
This function does more than just handle `reconnect`. The listeners for notices or errors should be added in the constructor.

Reply:
afaik you cannot reconnect an 'isomorphic-ws' , so this function is recreating the socket on reconnection, and for this reason it needs to add the listener to the new websocket there.

Reply:
Ah, I see. Makes sense then!